app/src/main/java/com/example/ncov19traking/ui/global/GlobalFragment.kt

```diff
@@ -64,28 +68,36 @@ class GlobalFragment : Fragment() {
     private fun setupObserverSubscription() {
         homeViewModel.nCoVAllCases.observe(viewLifecycleOwner, Observer {
             with(GlobalFragmentViewHolder()) {
-                totalCases.text = it.cases.toString()
-                recovered.text = it.recovered.toString()
-                deaths.text = it.deaths.toString()
+                covidTotalCasesCard.setCaseNumbers(formatLargeNumbers(it.cases))
+                covidRecoveredCard.setCaseNumbers(formatLargeNumbers(it.recovered))
+                covidDeathsCard.setCaseNumbers(formatLargeNumbers(it.deaths))
                 lastUpdateLong.text = Date(it.updated).toString()
-                casesPercentage.text = getString(
-                    R.string.since_yesterday, homeViewModel.getCasesPercentageDifference(
-                        it.cases,
-                        homeViewModel.nCoVYesterdayAllCases.cases
+                covidTotalCasesCard.setPercentageText(
+                    getString(
+                        R.string.since_yesterday, homeViewModel.getCasesPercentageDifference(
+                            it.cases,
+                            homeViewModel.nCoVYesterdayAllCases.cases
+                        )
                     )
                 )
-                recoveredPercentage.text = getString(
-                    R.string.since_yesterday, homeViewModel.getCasesPercentageDifference(
-                        it.recovered,
-                        homeViewModel.nCoVYesterdayAllCases.recovered
+                covidRecoveredCard.setPercentageText(
+                    getString(
+                        R.string.since_yesterday, homeViewModel.getCasesPercentageDifference(
+                            it.recovered,
+                            homeViewModel.nCoVYesterdayAllCases.recovered
+                        )
                     )
                 )
-                deathsPercentage.text = getString(
-                    R.string.since_yesterday, homeViewModel.getCasesPercentageDifference(
-                        it.deaths,
-                        homeViewModel.nCoVYesterdayAllCases.deaths
+                covidDeathsCard.setPercentageText(
+                    getString(
+                        R.string.since_yesterday, homeViewModel.getCasesPercentageDifference(
+                            it.deaths,
+                            homeViewModel.nCoVYesterdayAllCases.deaths
+                        )
                     )
                 )
+                pieChart.getChartData(it)
+                pieChart.setVisibility(true)
                 progressBar.isVisible = false
             }
```

Comment:
this whole block could be separated in different methods to make it more readable, I see three different things being done, setting percentages, configuring piechart, and setting case numbers, try to be careful with this things too make your code more organized.